meta-lxatac-software: update various software packages

hnez opened the pull request:
This PR updates multiple software packages that we package in meta-lxatac to their most recent release:

  - bottom
  - github-act-runner
  - gitlab-runner
  - py-spy
  - ripgrep
  - imx-uuu

---

TODO before merging:

- [x] Since this is a PR into the new scarthgap branch, that does not yet contain scarthgap-y content #137 should be merged first, which performs the scarthgap update.

hnez committed:
meta-lxatac-software: bottom: update to version 0.9.6

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: github-act-runner: update to version 0.6.10

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: gitlab-runner: update to version 16.6.1

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: py-spy: update to version 0.3.14

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: ripgrep: update to version 14.1.0

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: imx-uuu: update to version 1.5.165

In the meantime uuu has started to depend on tinyxml2.
Since tinyxml2 was not packaged yet a recipe for it was added as well.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented on meta-lxatac-software/recipes-rust/py-spy/py-spy_0.3.14.bb:
The version number is admittedly quite cool for a py project, but staying at it since 2022 has some drawbacks, namely:

```bash
root@lxatac-00011:~# py-spy --version
py-spy 0.3.14
root@lxatac-00011:~# python3 -V
Python 3.12.3
root@lxatac-00011:~# py-spy record --pid 702
Error: Unsupported version of Python: 3.12.0
```

Should we keep `py-spy` even though it is not usable on the TAC?

SmithChart replied:
As far as I can tell no tooling on the TAC depends on `py-spy`.
When it is useless I vote to drop it.
Once it has been updated for the then-to-be most-recent Python, we can add it again.

jluebbe replied:
No other package depends on py-spy, as it's a command line tool for developers. I've used it to analyze performance of the labgrid-exporter. But if it no longer works with current the Python version, we should drop it. See https://github.com/benfred/py-spy/issues/633.

hnez replied:
I've added a commit that drops `py-spy` while keeping the commit that updates it to the most recent version so we have something to work with once a new version is released.

hnez committed:
meta-lxatac-software: py-spy: drop recipe

While py-spy is a useful tool it has sadly not seen a new release in some
time, which results in it no longer being compatible with the python
version we ship on the LXA TAC.
This means it does not make a lot of sense to ship py-spy right now.

We can re-enable the recipe once py-spy is updated to support current
python versions.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
I've had to disable `bottom` in the scarthgap update PR #137, because the old version does not build on new rust compilers.
The updated version we have here does not suffer from the same issue so we can re-enable it here.
To make sure I do not forget to do so I've cherry-picked the commit that disables `bottom` from #137 to here and re-enabled it right after. The former commit should go away when rebasing on top of a merged #137.

hnez committed:
meta-labgrid: lxa-iobus: update to version 0.5.1

Changelog for version 0.5.0:

 - Fix "LICENSE" typo in setup.cfg
 - Expose individual node info via REST and allow CORS requests
 - Add formatters and linters to the CI and fix the fallout
 - lxa_iobus: move the project from setup.py to pyproject.toml
 - Cleanups, code deduplication, improvements and maintenance
 - lxa_iobus: node: add HTTP API based LxaRemoteNode
 - Add support for the LXA Optick latency measurement util
 - lxa_iobus: firmware: update Ethernet Mux and 4DO-3DI-3AI firmware to 0.6.0
 - README: Update DCO to follow the kernel wording

Changelog for version 0.5.1:

 - lxa_iobus: server: fix string/integer type confusion in set_pin()

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-labgrid: usbsdmux: update to version 24.01.1

Changelog for version 24.01:

 - implement 'info' command to show SD card SCR, CID and CSD
 - Add JSON status output for all commands and errors
 - Add support for decoding SCR, CID and CSD registers
 - Add pytest tests and a GitHub Actions workflow for SD register decoding
 - add test that checks README.rst against usbsdmux -h output

Changelog for version 24.01.1:

 - Add statistics via mqtt
 - README: Update DCO to follow the kernel wording

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
Now that labgrid-project/meta-labgrid#48 and labgrid-project/meta-labgrid#49 we can also have an `lxa-iobus` server with [Optick](https://www.pengutronix.de/en/blog/2024-05-06-linux-automation-s-optick-glass-to-glass-latency-measurement-tool.html) support and a `usbsdmux` with MQTT based statistics reporting 🎉.

hnez commented:
Hi, now that #137 is merged and the dust has settled (default branches are updated, build jobs are updated and ran successfully, labgrid tests are updated and also ran fine and the first scarthgap based [testing bundle](https://downloads.linux-automation.com/lxatac/software/testing/2024-05-30/) is available and installable) we can go ahead and merge this PR as well..

hnez commented:
I've just noticed that this PR is still open. I think we should get this merged.
